src/App.tsx

```diff
@@ -15,48 +16,115 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortTypes {
+  SORT_FIELD_ALPHABETICALLY = 'alphabetically',
+  SORT_FIELD_BY_LENGTH = 'length',
+  SORT_REVERSE = 'reverse',
+  DEFAULT = '',
+}
+
+interface ISortButton {
+  name: string;
+  type: SortTypes;
+  style: string;
+}
+
+const sortButtons: ISortButton[] = [
+  {
+    name: 'Sort alphabetically',
+    type: SortTypes.SORT_FIELD_ALPHABETICALLY,
+    style: 'is-info',
+  },
+  {
+    name: 'Sort by length',
+    type: SortTypes.SORT_FIELD_BY_LENGTH,
+    style: 'is-success',
+  },
+  {
+    name: 'Reverse',
+    type: SortTypes.SORT_REVERSE,
+    style: 'is-warning',
+  },
+];
+
+function getPreparedGoods(
+  goods: string[],
+  sortField: typeof SortTypes[keyof typeof SortTypes],
+  isReversed: boolean,
+) {
+  const preparedGoods = [...goods];
+
+  if (sortField === SortTypes.SORT_FIELD_ALPHABETICALLY) {
+    preparedGoods.sort((a, b) => a.localeCompare(b));
+  }
+
+  if (sortField === SortTypes.SORT_FIELD_BY_LENGTH) {
+    preparedGoods.sort((a, b) => a.length - b.length);
+  }
+
+  if (isReversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField]
+    = useState<typeof SortTypes[keyof typeof SortTypes]>(SortTypes.DEFAULT);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const visibleGoods = getPreparedGoods(
+    goodsFromServer, sortField, isReversed,
+  );
+
+  const handleSort = (type: typeof SortTypes[keyof typeof SortTypes]) => {
+    if (type === SortTypes.SORT_REVERSE) {
+      setIsReversed(!isReversed);
+    } else {
+      setSortField(type);
+    }
+  };
+
+  const handleReset = () => {
+    setSortField(SortTypes.DEFAULT);
+    setIsReversed(false);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
+        {sortButtons.map(({ name, type, style }) => (
+          <button
+            key={type}
+            type="button"
+            className={cn('button', style, {
+              'is-light': type === SortTypes.SORT_REVERSE
+                ? !isReversed
+                : type !== sortField,
+            })}
+            onClick={() => handleSort(type)}
+          >
+            {name}
+          </button>
+        ))}
```

Comment:
i maight be wrong but in such cases i wouldn't use cycle because it makes code unclear and also it would be difficult to change behavior of a specific button. Also it would be really messy if we add some buttons with new behaviour.

And also it's still ok in cases such as in line 126, but here we have a big block of code, so if you want to make a cycle - i would recommend you to make a separate class and pass props to it.